File: articles/three-good-things-exercise/fortran/appreciative_awareness_index.f90
program appreciative_awareness_index
  implicit none
  real :: life_satisfaction, depressive_symptoms, gratitude_score, positive_event_salience
  real :: perceived_support, reflection_depth, stress_load, acceptability, context_fit
  real :: appreciative_awareness, net_wellbeing

  life_satisfaction = 7.4
  depressive_symptoms = 3.6
  gratitude_score = 7.2
  positive_event_salience = 7.1
  perceived_support = 7.3
  reflection_depth = 7.1
  stress_load = 3.2
  acceptability = 7.7
  context_fit = 7.6

  appreciative_awareness = (gratitude_score + positive_event_salience + perceived_support + &
      reflection_depth + acceptability + context_fit - stress_load) / 7.0

  net_wellbeing = life_satisfaction + gratitude_score + positive_event_salience + &
      perceived_support + reflection_depth - depressive_symptoms - stress_load

  print *, "Appreciative awareness index:", appreciative_awareness
  print *, "Net wellbeing index:", net_wellbeing
end program appreciative_awareness_index

File: articles/three-good-things-exercise/c/appreciative_awareness_index.c
#include <stdio.h>

typedef struct {
    double life_satisfaction;
    double depressive_symptoms;
    double gratitude_score;
    double positive_event_salience;
    double perceived_support;
    double reflection_depth;
    double stress_load;
    double acceptability;
    double context_fit;
} ThreeGoodThingsIndicators;

double appreciative_awareness_index(ThreeGoodThingsIndicators x) {
    return (
        x.gratitude_score +
        x.positive_event_salience +
        x.perceived_support +
        x.reflection_depth +
        x.acceptability +
        x.context_fit -
        x.stress_load
    ) / 7.0;
}

double net_wellbeing_index(ThreeGoodThingsIndicators x) {
    return x.life_satisfaction
         + x.gratitude_score
         + x.positive_event_salience
         + x.perceived_support
         + x.reflection_depth
         - x.depressive_symptoms
         - x.stress_load;
}

int main(void) {
    ThreeGoodThingsIndicators example = {7.4, 3.6, 7.2, 7.1, 7.3, 7.1, 3.2, 7.7, 7.6};
    printf("Appreciative awareness index: %.3f\n", appreciative_awareness_index(example));
    printf("Net wellbeing index: %.3f\n", net_wellbeing_index(example));
    return 0;
}
